apps/openmw/mwgui/levelupdialog.cpp

```diff
@@ -248,4 +250,103 @@ namespace MWGui
+    {
+        std::string ret = "acrobat";
+
+        int total = combatIncreases + magicIncreases + stealthIncreases;
+        if (total == 0)
+            return ret;
+
+        int combatFraction = static_cast<int>(static_cast<float>(combatIncreases) / total * 10.f);
+        int magicFraction = static_cast<int>(static_cast<float>(magicIncreases) / total * 10.f);
+        int stealthFraction = static_cast<int>(static_cast<float>(stealthIncreases) / total * 10.f);
+
+        if (combatFraction > 7)
+            ret = "warrior";
+        else if (magicFraction > 7)
+            ret = "mage";
+        else if (stealthFraction > 7)
+            ret = "thief";
+
+        switch (combatFraction)
+        {
+            case 7:
+                ret = "warrior";
+                break;
+            case 6:
+                if (stealthFraction == 1)
+                    ret = "barbarian";
+                else if (stealthFraction == 3)
+                    ret = "crusader";
+                else
+                    ret = "knight";
+                break;
+            case 5:
+                if (stealthFraction == 3)
+                    ret = "scout";
+                else
+                    ret = "archer";
+                break;
+            case 4:
+                ret = "rogue";
+                break;
+            default:
+                break;
+        }
+
+        switch (magicFraction)
+        {
+            case 7:
+                ret = "mage";
+                break;
+            case 6:
+                if (combatFraction == 2)
+                    ret = "sorcerer";
+                else if (combatIncreases == 3)
+                    ret = "healer";
+                else
+                    ret = "battlemage";
+                break;
+            case 5:
+                ret = "witchhunter";
+                break;
+            case 4:
+                ret = "spellsword";
+                // In vanilla there's also code for "nightblade", however it seems to be unreachable.
+                break;
+            default:
+                break;
+        }
+
+        switch (stealthFraction)
+        {
+            case 7:
+                ret = "thief";
+                break;
+            case 6:
+                if (magicFraction == 1)
+                    ret = "agent";
+                else if (magicIncreases == 3)
+                    ret = "assassin";
+                else
+                    ret = "acrobat";
+                break;
+            case 5:
+                if (magicIncreases == 3)
+                    ret = "monk";
+                else
+                    ret = "pilgrim";
+                break;
+            case 3:
+                if (magicFraction == 3 && stealthFraction == 3)
+                    ret = "bard";
+                break;
+            default:
+                break;
+        }
+
+        return ret;
+    }
 }
```

Comment:
Another thing, which I noted in the bug report (I'm using the name "R.D." there) is that only skills that are advanced through actual use are counted, not skills gained through training or books. I'm not sure why the developers might have not wanted to allow training or books skills to have an effect, as to me they seem as much like a true part of the character as skills gained through actual use. That also might be an oversight.

Well, I don't know what the official OpenMW policy is on implementation is in situations like this, maybe to just stick to the original behavior. My own preference is to go for what looks most likely to have been the original intent, although that introduces the problem of interpretation. My preference would be, while documenting the original behavior in the source code or somewhere:
- use fractions in place of the increases
- count advances made through books and training
- implement nightblade (something like if stealthfraction = 3 seems like a likely candidate)
- reset the count on leveling

However I think you've already done great work and I don't mean to make any kind of demands of you. As you say this has little impact and it's not a big deal if you just want to be done with it.